tests/AC_model_manager/test_model_download.py

```diff
@@ -0,0 +1,361 @@
+        job = queue.create_download_job(source="http://www.civitai.com/models/malicious", destdir=tmpdir)
+        queue.join()
+        assert job.status == "completed"
+        assert job.destination == Path(tmpdir, "malicious")
+        assert Path(tmpdir, "malicious").exists()
+
+        # Nor a destination that would exceed the maximum filename or path length
+        job = queue.create_download_job(source="http://www.civitai.com/models/long", destdir=tmpdir)
+        queue.join()
+        assert job.status == "completed"
+        assert job.destination == Path(tmpdir, "long")
+        assert Path(tmpdir, "long").exists()
+
+    # create a foreign job which will be invalid for the queue
+    bad_job = DownloadJobBase(id=999, source="mock", destination="mock")
+    try:
+        queue.start_job(bad_job)  # this should fail
+        succeeded = True
+    except UnknownJobIDException:
+        succeeded = False
+    assert not succeeded
+
+
+def test_pause_cancel_url():  # this one is tricky because of potential race conditions
+    def event_handler(job: DownloadJobBase):
+        time.sleep(0.5)  # slow down the thread so that we can recover the paused state
+
+    queue = ModelDownloadQueue(requests_session=session, event_handlers=[event_handler])
+    with tempfile.TemporaryDirectory() as tmpdir:
+        job1 = queue.create_download_job(source="http://www.civitai.com/models/12345", destdir=tmpdir, start=False)
+        job2 = queue.create_download_job(source="http://www.civitai.com/models/9999", destdir=tmpdir, start=False)
+        job3 = queue.create_download_job(source="http://www.civitai.com/models/54321", destdir=tmpdir, start=False)
+
+        assert job1.status == "idle"
+        queue.start_job(job1)
+        queue.start_job(job3)
+        time.sleep(0.1)  # wait for enqueueing
+        assert job1.status in ["enqueued", "running"]
+
+        # check pause and restart
+        queue.pause_job(job1)
+        time.sleep(0.1)  # wait to be paused
+        assert job1.status == "paused"
+
+        queue.start_job(job1)
+        time.sleep(0.1)
+        assert job1.status == "running"
+
+        # check cancel
+        queue.start_job(job2)
+        time.sleep(0.1)
+        assert job2.status == "running"
+        queue.cancel_job(job2)
+        time.sleep(0.1)
+        assert job2.status == "cancelled"
+
+        queue.join()
+        assert job1.status == "completed"
+        assert job2.status == "cancelled"
+        assert job3.status == "completed"
+
+        assert Path(tmpdir, "mock12345.safetensors").exists()
+        assert Path(tmpdir, "mock9999.safetensors").exists() is False, "cancelled file should be deleted"
+        assert Path(tmpdir, "mock54321.safetensors").exists()
+
+        assert len(queue.list_jobs()) == 3
+        queue.prune_jobs()
+        assert len(queue.list_jobs()) == 0
+
+    def test_pause_cancel_repo_id():  # this one is tricky because of potential race conditions
```

Comment:
Fix the indentation here. This test will never run.

Reply:
fixed